### PR TITLE
Performs a huge refactoring: Users no longer have an ID, only a usern…

### PR DESCRIPTION
**Pull request by lancelafontaine**
…ame; Timeslots and Reservations are now identified with a 128-bit random string; Timeslot ID and Timeslot User Id are no longer mistakenly interchanged everywhere; Updates all unit tests; Controller is now ble to fetch the appropriate timeslot for a given reservation and return it to the client-side

### CODE REVIEW

File: app/core/reservationbook_test.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def mockWaitingMapperDone():
 	assert len(reservationBook.reservationList) == 1
 
 	"""Making sure the user is the first capstone student at the top of the list"""
-	assert reservationBook.reservationList[0].getUser().getName() == 'emir'
+	assert reservationBook.reservationList[0].getUser().getId() == 'emir'
```
Comment (adriannadiaz) [inline app/core/reservationbook_test.py]: Don't we want `getUsername()` ??

File: app/core/reservationbook_test.py [hunk 2]
```diff
@@ -131,7 +131,7 @@ def mockWaitingMapperDone():
 	assert len(reservationBook.reservationList) == 1
 
 	"""Making sure that the user in the reservation list is Mark"""
-	assert reservationBook.reservationList[0].getUser().getName() == "mark"
+	assert reservationBook.reservationList[0].getUser().getId() == "mark"
```
Comment (adriannadiaz) [inline app/core/reservationbook_test.py]: Same here

File: app/mapper/UserMapper_test.py [hunk 3]
```diff
@@ -26,14 +26,14 @@ def no_find2(_, __):
...
-    expected = User(10, 'joe', 'shmoe')
+    expected = User('joe', 'shmoe')
 
     # Mock
     def no_find(_, __):
         return
 
     def yes_find(_):
-        return [[expected.getId(), expected.getName(), expected.getPassword(), expected.isCapstone()]]
+        return [[expected.getId(), expected.getPassword(), expected.isCapstone()]]
```
Comment (adriannadiaz) [inline app/mapper/UserMapper_test.py]: This getId() is odd, I thought it was `getUsername()`

File: app/core/user.py [hunk 4]
```diff
@@ -5,31 +5,24 @@ class User:
...
+    def __init__(self, username, password, capstone=False):
+        self.username = username
         self.password = password
-        self.userId = userId
         self.capstone = capstone
 
     # Accessors and Mutators
-    def getName(self):
-        return self.name
+    def getId(self):
```
Comment (adriannadiaz) [inline app/core/user.py]: It is `getUsername()`

File: app/core/user.py [hunk 5]
```diff
@@ -5,31 +5,24 @@ class User:
...
+        return self.username
 
-    def setName(self, name):
-        self.name = name
+    def setId(self, username):
```
Comment (adriannadiaz) [inline app/core/user.py]: It should be `setUsername()`

File: app/mapper/TimeslotMapper_test.py [hunk 6]
```diff
@@ -98,3 +99,4 @@ def tdg_find(_):
     assert(val.getDate() is expected.getDate())
     assert(val.getBlock() is expected.getBlock())
     assert(val.getId() is expected.getId())
+    assert(val.getUserId() is expected.getUserId())
```
Comment (adriannadiaz) [inline app/mapper/TimeslotMapper_test.py]: `getUsername()`

File: app/core/reservationbook_test.py (hunk 1 above)
Comment (lancelafontaine) [inline app/core/reservationbook_test.py]: We can really do this, because it will break how Aspects work. They depend on `getId()` being available.

**Pull request merged**